src/test/java/htmlpublisher/HtmlFileNameTest.java

```diff
@@ -1,32 +1,39 @@
 package htmlpublisher;
 
-import static org.junit.Assert.assertEquals;
+import java.util.List;
 
-import java.util.Arrays;
-
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 import org.jvnet.hudson.test.CreateFileBuilder;
 import org.jvnet.hudson.test.JenkinsRule;
 
 import org.htmlunit.html.HtmlInlineFrame;
 import org.htmlunit.html.HtmlPage;
 
 import hudson.model.FreeStyleProject;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@WithJenkins
+class HtmlFileNameTest {
 
-public class HtmlFileNameTest {
-    @Rule
-    public JenkinsRule j = new JenkinsRule();
+    private JenkinsRule j;
+
+    @BeforeEach
+    void setUp(JenkinsRule rule) {
```

Comment:
I didn't look at how the junit5 extension is done, but it looks like  something which should be done automatically or just by adding some annotations to the `JenkinsRule j` field

Reply:
While I do agree that it would be more convenient that way, it is not (yet) working that way: See https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/junit/jupiter/WithJenkins.html